src/App.tsx

```diff
@@ -8,37 +8,117 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getPosts, getUsers } from './api/services';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const selectUserHandler = (user: User) => {
+    setSelectedUser(user);
+  };
+
+  const selectPostHandler = (post: Post) => {
+    setSelectedPost(post);
+  };
+
+  const loadUserFromServer = async () => {
```

Comment:
Would be great to create async functions in useEffect
https://devtrium.com/posts/async-functions-useeffect

Reply:
Thanks! Looks better now